### ISSUE TITLE
I am unable to use bundlerEnv to create an environment that includes tzinfo

### ISSUE THREAD

**Issue by savannidgerinel** [NixOS/nixpkgs#66411]
**Describe the bug**
I have created a shell environment for Ruby development. It includes Ruby 2.6.3p62, Bundler 1.17.2, and Bundix 2.4.1, all from the Unstable channel on 2019-08-09). If I include tzinfo 1.2.5 or tzinfo 2.0.0, even as a transient dependency, tzinfo fails to build and the environment fails to instantiate.

I have created a minimal shell.nix and supporting files at https://github.com/savannidgerinel/nixpkgs-ruby-tzinfo.

The error message is:

```
warning: unknown setting 'signed-binary-caches'
these derivations will be built:
  /nix/store/kx2gsl3c9ciryal6x7pi4clj7d5hd2y1-ruby2.6.3-tzinfo-1.2.5.drv
  /nix/store/13k5s8x7cjbmnvarf9423zrgc244wr0g-tzinfo-demo-env.drv
warning: unknown setting 'signed-binary-caches'
building '/nix/store/kx2gsl3c9ciryal6x7pi4clj7d5hd2y1-ruby2.6.3-tzinfo-1.2.5.drv'...
unpacking sources
Unpacked gem: '/private/var/folders/13/84dj8yr54_1c_pn0s8n7444h0000gn/T/nix-build-ruby2.6.3-tzinfo-1.2.5.drv-0/container/mg5gy5dz5rxbg73wkkrgx3vszqxya1ib-tzinfo-1.2.5'
patching sources
configuring
no configure script, doing nothing
building
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
ERROR:  While executing gem ... (Gem::InvalidSpecificationException)
    [".yardopts", "CHANGES.md", "LICENSE", "README.md", "Rakefile", "lib/tzinfo.rb",
    ...
    "test/zoneinfo/iso3166.tab", "test/zoneinfo/leapseconds", "test/zoneinfo/posix/Europe/London", "test/zoneinfo/posixrules", "test/zoneinfo/right/Europe/London", "test/zoneinfo/zone.tab", "test/zoneinfo/zone1970.tab", "tzinfo.gemspec"] are not files
builder for '/nix/store/kx2gsl3c9ciryal6x7pi4clj7d5hd2y1-ruby2.6.3-tzinfo-1.2.5.drv' failed with exit code 1
cannot build derivation '/nix/store/13k5s8x7cjbmnvarf9423zrgc244wr0g-tzinfo-demo-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/13k5s8x7cjbmnvarf9423zrgc244wr0g-tzinfo-demo-env.drv' failed
```

**To Reproduce**
1. Ensure the nixpkgs channel points to https://nixos.org/channels/nixpkgs-unstable (although this also happens on https://nixos.org/channels/nixpkgs-19.03-darwin)
2. Check out the repository https://github.com/savannidgerinel/nixpkgs-ruby-tzinfo
3. Run nix-shell at the root of the repository

**Expected behavior**
I expect to get a shell environment that has Ruby and in which I can import tzinfo.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

```
warning: unknown setting 'signed-binary-caches'
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
warning: unknown setting 'signed-binary-caches'
 - version: `nix-env (Nix) 2.1.3`
warning: unknown setting 'signed-binary-caches'
 - channels(savanni): `"nixpkgs-19.09pre187222.002b853782e, nixpkgs-18.09-18.09pre154169.68b3bff32da, nixpkgs-19.03-19.03pre173257.f8096b25f80"`
warning: unknown setting 'signed-binary-caches'
 - nixpkgs: `/Users/savanni/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by worldofpeace** [comment]
cc @alyssais @manveru

**Comment by thefloweringash** [comment]
Similar symptoms to https://github.com/NixOS/nixpkgs/issues/60676.

**Comment by alyssais** [comment]
Would be fixed by either #61684 or https://github.com/NixOS/nixpkgs/pull/61114.

**Comment by alyssais** [comment]
This is now fixed in staging so I’m going to close this.
